core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -92,10 +107,17 @@ public void setEnableExpressionCache(String cache) {
 
     @Inject(value = StrutsConstants.STRUTS_ENABLE_OGNL_EVAL_EXPRESSION, required = false)
     public void setEnableEvalExpression(String evalExpression) {
-        enableEvalExpression = "true".equals(evalExpression);
-        if(enableEvalExpression){
-            LOG.warn("Enabling OGNL expression evaluation may introduce security risks " +
-                    "(see http://struts.apache.org/release/2.3.x/docs/s2-013.html for further details)");
+        if (!this.enableEvalExpressionValueSet) {
```

Comment:
Sorry you did a lot of work but please replace all change-limiters logic including their unit test with proposed solution above. Thanks in advance!